code/code/misc/being.h

```diff
@@ -1404,6 +1404,7 @@ class TBeing : public TThing {
   bool checkSmashed(TBeing *, wearSlotT, spellNumT, TThing *, int, const char * = NULL);
   int hit(TBeing *, int pulse = -1);
   bool canCounterMove(int);
+  bool canFocusedAvoidance(int);
```

Comment:
I wonder why these skills live in TBeing? This causes horrible recompilation times for everything, as well as leaking inner workings of disciplines into generic code of beings.

As most things in TBeing are public, maybe it makes more sense to have that code in disciplines?

Reply:
Seems like all the combat type code is dealing with two objects that are subclasses of TBeing of some sort. So it kind of makes sense to be able to write 
```
if (victim->canFocusedAvoidance(bKnown/2)) {
```
I guess the other option would just be

```
if (canFocusedAvoidance(victim, bKnown/2)) {
```
Just not object oriented. It seems a little easier to reason about in the first instance but if the performance impact is huge the other way it's not a big deal to change it all.

I feel like "Disciplines" or "Abilities" maybe should be a class that is instantiated per being.